app/models/king.rb

```diff
@@ -1,15 +1,98 @@
 class King < Piece
+
+  def y_position
```

Comment:
Hmm what about the existing `y_position` attribute in the database? Seems that this would override that in instances where castling is not happening?